File: src/001.Factory/simple-factory/ShapeFactory.java
import Circle;
import Shape;

public class ShapeFactory{
    public static Shape getShape(String type){
        if(type.equals("Circle")){
            return new Circle();
        }
        if(type.equals("Square")){
            return new Square();
        }
        throw new RuntimeException("Bad Shape creation: "+type);
    }
}

File: src/001.Factory.cpp/src/ShapeFactory.cpp
#include<iostream>
#include<string>

using namespace std;

enum TYPE{CIRCLE, SQUARE};

class Shape
{
	public:
		virtual void show()=0;
};

class Circle: public Shape
{
	public:
		void show()
		{
			cout << "Circle" << endl;
		}
};

class Square: public Shape
{
	public:
		void show()
		{
			cout << "Square" << endl;
		}
};

class ShapeFactory
{
	public:
		Shape* getShape(enum TYPE type)
		{
			if(type == CIRCLE)
			{
				return new Circle();
			}
			if(type == SQUARE)
			{
				return new Square();
			}
		}
};
